source/shortcut.h

```diff
@@ -0,0 +1,15 @@
+#ifndef INC_SHORTCUT_H
+#define INC_SHORTCUT_H
+#include <gctypes.h>
+
+#define NUM_SHORTCUTS 6
```

Comment:
You're changing this from calculating the length to hardcoding a value. This is easy to miss and forget updating. Just export the symbol from the new file.